### PR TITLE
Add iOS project to empty game template

### PR DESCRIPTION
**Pull request by TimOliver**
Up until now, the empty game template in osu!framework has only had build targets for desktop and the desktop test browser.

Using the build settings from the iOS sample game in this repo (Which was the basis for flappy-osu) as well as examining some of the build settings in osu!stream, this PR adds the ability to build and deploy to iOS and the iOS Simulator.

### CODE REVIEW

File: osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    <MtouchArch>ARM64</MtouchArch>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|iPhoneSimulator' ">
+    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <MtouchLink>SdkOnly</MtouchLink>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|iPhone' ">
+    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <CodesignKey>Apple Development: Timothy Oliver (RSQSFECVR6)</CodesignKey>
```
Comment (swoolcock) [inline osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj]: Maybe don't include your own code signing info. 😅

Comment (TimOliver) [reply osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj]: Hahaha fair point. 🤣

Comment (bdach) [reply osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj]: I don't know how sensitive/confidential this info is, but *if* it is, I'd argue this is an actual case where you want to rebase it out and force-push to purge history entirely (Google tells me commits on PRs are GC'd after force). Unless that key is already revoked or something.

Comment (TimOliver) [reply osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj]: Nah. It’s not confidential at all. 🙂

No one can do anything with that without the accompanying signing certificate. It doesn’t need to be removed from the git history.

But thanks for the heads-up! 😄

File: osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj [hunk 2]
```diff
@@ -64,6 +64,7 @@
     <MtouchEnableSGenConc>true</MtouchEnableSGenConc>
     <MtouchLink>None</MtouchLink>
     <MtouchExtraArgs>--nolinkaway -gcc_flags "-lstdc++ -lbz2 -framework AudioToolbox -framework AVFoundation -framework CoreMedia -framework VideoToolbox -framework SystemConfiguration -framework CFNetwork -framework Accelerate"</MtouchExtraArgs>
+    <CodesignKey>iPhone Developer</CodesignKey>
```
Comment (smoogipoo) [inline osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj]: Can these be removed completely instead? Just the ones added in this commit.

File: osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    <MtouchLink>None</MtouchLink>
+    <MtouchExtraArgs>--nolinkaway -gcc_flags "-lstdc++ -lbz2 -framework AudioToolbox -framework AVFoundation -framework CoreMedia -framework VideoToolbox -framework SystemConfiguration -framework CFNetwork -framework Accelerate"</MtouchExtraArgs>
+    <CodesignKey>iPhone Developer</CodesignKey>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Testing|iPhone' ">
+    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <DebugType>none</DebugType>
+    <Optimize>True</Optimize>
+    <OutputPath>bin\iPhone\Testflight</OutputPath>
+    <DefineConstants>iOS RELEASE MAPPER</DefineConstants>
```
Comment (peppy) [inline osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj]: these are still showing through from your... donor of this file. i'd recommend generating a fresh csproj (using a xamarin template) and using it as much as possible.

File: osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj (hunk 2 above)
Comment (TimOliver) [reply osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj]: I just tried removing them, Visual Studio just put them back in. They're used as a guideline for which certificate to build for this particular target.

File: osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj (hunk 3 above)
Comment (TimOliver) [reply osu.Framework.Templates/templates/template-empty/TemplateGame.iOS/TemplateGame.iOS.csproj]: Done. :D

**Comment by peppy** [comment]
The icon resources need to be replaced or removed (they are using the osu! icon right now).

**Comment by TimOliver** [comment]
@peppy Okay! The artwork's been removed, and I went through every build combination that I could to make sure both Desktop and iOS configurations were working properly. :)

File: osu.Framework.Templates/templates/template-empty/TemplateGame.Game/TemplateGame.Game.csproj [hunk 4]
```diff
@@ -1,11 +1,13 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>netstandard2.1</TargetFramework>
+    <Configurations>Debug;Release</Configurations>
```
Comment (smoogipoo) [inline osu.Framework.Templates/templates/template-empty/TemplateGame.Game/TemplateGame.Game.csproj]: This shouldn't be in a netstandard2.1 project file (can just be straight-up removed).

```suggestion
```

File: osu.Framework.Templates/templates/template-empty/TemplateGame.Game/TemplateGame.Game.csproj [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
+    <Platforms>AnyCPU</Platforms>
```
Comment (smoogipoo) [inline osu.Framework.Templates/templates/template-empty/TemplateGame.Game/TemplateGame.Game.csproj]: neither

```suggestion
```

Comment (TimOliver) [reply osu.Framework.Templates/templates/template-empty/TemplateGame.Game/TemplateGame.Game.csproj]: Done!

**Comment by peppy** [comment]
Tested full deploy process using fresh template to iOS device and simulator; both work flawlessly 👍

**Pull request merged**